src/xopen/__init__.py

```diff
@@ -85,23 +86,18 @@ def __del__(self):
             pass
 
 
-class PipedGzipWriter(Closing):
+class PipedCompressionWriter(Closing):
     """
-    Write gzip-compressed files by running an external gzip or pigz process and
-    piping into it. pigz is tried first. It is fast because it can compress using
-    multiple cores.
-
-    If pigz is not available, a gzip subprocess is used. On Python 2, this saves
-    CPU time because gzip.GzipFile is slower. On Python 3, gzip.GzipFile is on
-    par with gzip itself, but running an external gzip can still reduce wall-clock
-    time because the compression happens in a separate process.
+    Write Compressed files by running an external process and piping into it.
     """
-
-    def __init__(self, path, mode='wt', compresslevel=6, threads=None):
+    def __init__(self, path, program, mode='wt',
+                 compresslevel: Optional[int] = None,
+                 threads_flag: str = None,
+                 threads: Optional[int] = None):
```

Comment:
It’s ok to leave this now, but adding type annotation is somewhat unrelated and should have been a separate PR (also because it’s not done consistently in all changed places). Just leave it as it is now.